frontend/src/features/swap/hooks/eth/use-handle-eth-submit.ts

```diff
@@ -0,0 +1,62 @@
+import { HexString } from '@gear-js/api';
+import { useMutation } from '@tanstack/react-query';
+import { useConfig, useWriteContract } from 'wagmi';
+import { watchContractEvent } from 'wagmi/actions';
+
+import { isUndefined } from '@/utils';
+
+import { BRIDGING_PAYMENT_ABI, ETH_BRIDGING_PAYMENT_CONTRACT_ADDRESS } from '../../consts';
+import { FormattedValues } from '../../types';
+
+import { useApprove } from './use-approve';
+
+function useHandleEthSubmit(ftAddress: HexString | undefined, fee: bigint | undefined, allowance: bigint | undefined) {
```

Comment:
Why not simplify the function signature like this? 

`function useHandleEthSubmit(ftAddress?: HexString, fee?: bigint, allowance?: bigint) {`

Is there a reason to prefer the current approach?

Reply:
Since `IntelliSense` is silent if parameters are optional, it would be easy to skip some of them during `useHandleSubmit()` call. For example, `useHandleSubmit('0x00', 0n)` (`allowance` is missing) would be a valid input that is annoying to debug.